Add blog post: LibrePCB 0.1.1 Released (#29)
LibrePCB/LibrePCB.github.io: content/blog/2019-03-31_release_0.1.1.md

---
date: 2019-03-31
title: LibrePCB 0.1.1 Released
author: U. Bruhin
---

### Highlights

#### Design Rule Check for Library Elements ([#381](https://github.com/LibrePCB/LibrePCB/pull/381))

> In probably every EDA tool it's easy to make mistakes when creating library

Reviewer: ```suggestion
In virtually every EDA tool it's easy to make mistakes when creating library
```

elements like symbols or packages. But most tools don't show warnings about
such mistakes right in the library editor, usually you notice them later in the

> schematic or board editors because of the ERC or DRC.

Reviewer: Maybe add a paragraph like the following?

> Many projects try to solve this problem using documentation and design guidelines. But humans are generally lazy  and don't like to read those guidelines. And even when new contributors read design guidelines, they usually cannot memorize everything, leading to a lot of review effort for maintainers and frustration for contributors when submitting newly created elements to the standard library.

Author: IMHO the blog post is already pretty long, thus I don't like to add one more whole paragraph ;) But I like the first two sentences and thus tried to integrate them into the existing paragraphs in cffd0bdae7a9b611f1866930743f7857d4ceeb71. What do you think about this?

Reviewer: Looks good!

> LibrePCB 0.1.1 now shows library element issues right in the library editor,

Reviewer: ```suggestion
LibrePCB 0.1.1 now shows potential problems with library elements right in the library editor,
```

so you can fix them immediately. And even better, for many of the warnings, the
library editor provides the corresponding fix with just one click! Let's see how
it works:

> [![Library Element Checks](/img/library_editor_rulecheck.gif)](/img/library_editor_rulecheck.gif)<br/><br/>

Author: Yes, this is expected ;) When building the site with Hugo it works properly...

#### More Powerful Symbol Variants ([#430](https://github.com/LibrePCB/LibrePCB/pull/430))

LibrePCB uses component symbol variants to support different graphical
representations for the same component, for example to distinguish between
American and European norms. See
[this talk at FOSDEM](https://www.youtube.com/watch?v=vu-h5y6tK34) (slides
[here](https://archive.fosdem.org/2018/schedule/event/cad_librepcb/attachments/slides/2267/export/events/attachments/cad_librepcb/slides/2267/librepcb_slides.pdf))
for details.

Although this was already implemented in LibrePCB 0.1.0, it was not very

> powerful yet because in the "Add Component"-dialog you had to select the

Reviewer: ```suggestion
powerful yet because in the "Add Component" dialog you had to select the
```

desired symbol variant manually every time. As typically the European symbol
variants were selected by default, it was especially annoying for Americans.

Now with the 0.1.1 release you can specify your preferred norm in the project

> settings, which will make the "Add Component"-dialog pre-select the symbol

Reviewer: ```suggestion
settings, which will make the "Add Component" dialog pre-select the symbol
```

variant which matches your preferred norm automatically! If you like to use
American symbols, just add the norm "IEEE 315" to your project settings:

[![Project Settings Dialog](/img/project_settings_dialog_norm.png)](/img/project_settings_dialog_norm.png)<br/><br/>

#### Transactional File System ([#422](https://github.com/LibrePCB/LibrePCB/pull/422))

This change is not directly visible to users, but it's still noteworthy because
it's a very fundamental refactoring of the file system access of libraries and
projects. The file system access is now transaction-based, which has several
advantages:

- More reliable save/autosave mechanism, making libraries and projects even more
  robust against file access issues or application crashes.
- Cleaner software architecture, leading to less error-prone code.

> - Much more future proof as it allows to implement features which were not

Reviewer: ```suggestion
- Much more future-proof as it allows to implement features which were not
```

  possible until now.

### Critical Bugfixes

Obround pads which are higher than wide (i.e. `height > width`) were rendered
wrong in the package and board editors of LibrePCB 0.1.0. In particular, their
orientation on screen was wrong by 90 degrees, while the Gerber output was
correct. This release fixes the wrong orientation, so if you have any pad
affected by this bug, it will now appear 90° rotated compared to the 0.1.0
release. See issue [#424](https://github.com/LibrePCB/LibrePCB/issues/424) for
details.

> 

Reviewer: I'd add a new section for standard library changes, since there were quite a few:

```
### Expanded Standard Library

Since the previous LibrePCB release quite a few noteworthy updates made it into the LibrePCB standard library:

- Added DFN package family
- Added LQFP/TQFP package family
- Added multiple density levels for SOIC package family
- Added packages for SOD123/SOD323/SOD523
- Added dual MOSFET components
- Updated SMD chip resistors (now with two density levels)
- ...and many other changes (including a few new vendor-specific libraries)

Thanks to the built-in library update mechanism, those changes are even available to users of the previous LibrePCB versions.

A lot of those new elements have been generated with the [librepcb-parts-generator](https://github.com/LibrePCB/librepcb-parts-generator/). Check it out if you're interested in adding new families of similar elements to the LibrePCB library!
```

Author: To be honest, I don't like to mention library updates in application release notes as these are two completely unrelated things. But I think it would be worth to write a separate blog post about the library updates. Would you be interested to write one? :)

Reviewer: Could do that, but then I'd like to tackle a few of the open issues first.

(Yes, the libraries are unrelated, but a release blogpost is also a bit of a "what's new since the last release" blogpost.)

Author: > Could do that, but then I'd like to tackle a few of the open issues first.

:+1:

I opened issue #30 for that (just to keep it in mind, it doesn't matter if the blog post is published in a few days or a few weeks).

- Make tabs closable with middle mouse button
  ([#388](https://github.com/LibrePCB/LibrePCB/pull/388))
- Add context menus to overview tab
  ([#380](https://github.com/LibrePCB/LibrePCB/pull/380),
  [#425](https://github.com/LibrePCB/LibrePCB/pull/425))
- Display element name in "save" action
  ([#419](https://github.com/LibrePCB/LibrePCB/pull/419))
- Disable menu items if not needed for the opened element
- Change default package line width to 0.2mm
  ([#413](https://github.com/LibrePCB/LibrePCB/pull/413))
- Sort component signal dropdown in device editor by name
  ([#427](https://github.com/LibrePCB/LibrePCB/pull/427))
- Fix possibly outdated information in component signal editor
  ([#398](https://github.com/LibrePCB/LibrePCB/pull/398))
- Start library rescan if library modified

#### Schematic Editor:
- Add project settings dialog
  ([#430](https://github.com/LibrePCB/LibrePCB/pull/430))
- Pre-select symbol variant in "Add Component"-dialog by norm
  ([#430](https://github.com/LibrePCB/LibrePCB/pull/430))

#### Board Editor:
- Fix empty view after removing a board
  ([#431](https://github.com/LibrePCB/LibrePCB/pull/431))

#### CLI:

> - Add support for opening *.lppz project files

Reviewer: ```suggestion
- Add support for opening `*.lppz` project files
```

- Make runnable on FreeBSD
  ([#417](https://github.com/LibrePCB/LibrePCB/pull/417))
- Remove backward compatibility to pre-0.1 file format
  ([#420](https://github.com/LibrePCB/LibrePCB/pull/420))
- Refactor file access with a transactional file system
  ([#422](https://github.com/LibrePCB/LibrePCB/pull/422))
- Refactor/improve workspace libraries handling
  ([#400](https://github.com/LibrePCB/LibrePCB/pull/400))
- Use separate *.sqlite files for each database version
  ([#401](https://github.com/LibrePCB/LibrePCB/pull/401))
- Load category trees from database instead of file system
  ([#421](https://github.com/LibrePCB/LibrePCB/pull/421))
- Filter all category trees to show only relevant entries
  ([#426](https://github.com/LibrePCB/LibrePCB/pull/426))
- Fix possible wrong orientation of obround pads
  ([#428](https://github.com/LibrePCB/LibrePCB/pull/428))
- Fix cell height issues of several tables on high DPI displays
  ([#429](https://github.com/LibrePCB/LibrePCB/pull/429))
- Add support for exporting projects to *.lppz archives
  ([#433](https://github.com/LibrePCB/LibrePCB/pull/433))

#### New languages:
- Polish (30% translated)
- Chinese (14% translated)

> - Spanish (0% translated)

Reviewer: Maybe add a link to Transifex?

Also, is it worth listing Spanish if it's at 0%?

Author: Added links to Transifex. I listed Spanish because of two reasons:
- It's now (automatically) listed in the workspace settings of LibrePCB
- When releasing, I take a look at the diff of translation files since the last release. If I don't list it now, I'll probably forget it in the next release blog post ;)

### Download

> The release can be downloaded for all major operating systems at our download

Reviewer: ```suggestion
The release can be downloaded for all major operating systems on our download
```
